### PR TITLE
migrate to null-safety

### LINKED ISSUE #22: Support for null-safety
Since Dart 2.12.0 beta landed in flutter beta channel, which brings sound null-safety it would be nice to have null-safety support on this package. This would enable everyone to start migrating their apps to null-safety. 

I also would like to work on this and will open an pull request tomorrow.

### PR DESCRIPTION
**Pull request by bierbaumtim**
fixes #22

### CODE REVIEW

File: lib/src/widget_builders.dart [hunk 1]
```diff
@@ -149,20 +153,20 @@ class ScreenTypeLayout extends StatelessWidget {
...
   const RefinedLayoutBuilder({
-    Key key,
+    Key? key,
     this.refinedBreakpoints,
     this.extraLarge,
     this.large,
-    @required this.normal,
-    this.small,
+    this.normal,
+    required this.small,
```
Comment (FilledStacks) [inline lib/src/widget_builders.dart:169]: Please make normal required and not small. the normal view is what you want to show if there's no UI defined.

File: lib/src/widget_builders.dart [hunk 2]
```diff
@@ -173,27 +177,27 @@ class RefinedLayoutBuilder extends StatelessWidget {
...
+          if (normal != null) return normal!(context);
           // If no normal layout is supplied we want to check if we have the size below it and display that
-          if (small != null) return small(context);
+          return small(context);
         }
 
-        // If none of the layouts above are supplied or we're on the normal size layout then we show the normal layout
-        return normal(context);
+        // If none of the layouts above are supplied or we're on the small size layout then we show the small layout
+        return small(context);
```
Comment (FilledStacks) [inline lib/src/widget_builders.dart:200]: Change back to using the normal function when there's nothing defined.

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Thanks for the PR. Just one thing that I'd like changed back and that's using the normal UI builder as the default builder instead of the small UI builder.

**Comment by bierbaumtim** [comment]
The problem is that if normal is required, the small function never gets used, because the normal function can't be null. So this will make the small function useless.
To solve the problem we should introduce a RefinedSize.small. With this we can check if the size is RefinedSize.small and than use the small one.

**Comment by FilledStacks** [comment]
You're right. Then we can instead update the case before the final return to check for small and then return the small() in there. Then we can leave the default if nothing is defined to return normal()

**Review by FilledStacks — APPROVED** [review]
Great!

**Comment by FilledStacks** [comment]
@bierbaumtim Thanks a lot for the help! I've published it.

**Pull request merged**